public/app/features/dashboard/components/DashboardSettings/AnnotationsSettings.tsx

```diff
@@ -1,31 +1,38 @@
-import React, { PureComponent } from 'react';
+import React, { useState } from 'react';
 import { DashboardModel } from '../../state/DashboardModel';
-import { AngularComponent, getAngularLoader } from '@grafana/runtime';
+import { AnnotationSettingsEdit, AnnotationSettingsList } from '../AnnotationSettings';
+import { newAnnotation } from '../AnnotationSettings/AnnotationSettingsEdit';
+import { DashboardSettingsHeader } from './DashboardSettingsHeader';
 
 interface Props {
   dashboard: DashboardModel;
 }
 
-export class AnnotationsSettings extends PureComponent<Props> {
-  element?: HTMLElement | null;
-  angularCmp?: AngularComponent;
+export const AnnotationsSettings: React.FC<Props> = ({ dashboard }) => {
+  const [editIdx, setEditIdx] = useState<number | null>(null);
 
-  componentDidMount() {
-    const loader = getAngularLoader();
+  const onGoBack = () => {
+    setEditIdx(null);
+  };
 
-    const template = '<div ng-include="\'public/app/features/annotations/partials/editor.html\'" />';
-    const scopeProps = { dashboard: this.props.dashboard };
-    this.angularCmp = loader.load(this.element, scopeProps, template);
-    this.angularCmp.digest();
-  }
+  const onNew = () => {
+    dashboard.annotations.list.push({
```

Comment:
Directly modifying props like this might cause some weird side effects, mostly that components depending on this prop won't detect the change as `dashboard.annotations.list` identity stays the same.

Reply:
yes, this should be done immutably